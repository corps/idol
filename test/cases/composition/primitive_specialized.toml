[SpecializedInt]
is_a = "int"
tags = ["thing"]

[UnspecializedInt]
is_a = ["int", "primitive_widens.IntAlias"]
tags = ["description:hello"]

[SpecializedString]
is_a = "string"
tags = ["thing"]

[SpecializedCovariantInt]
is_a = ["UnspecializedInt", "SpecializedInt"]

[SpecializedContravariantInt]
is_a = ["UnspecializedInt", "SpecializedInt"]
variance = "Contravariant"

[TwoSpecializedContravariant]
is_a = ["SpecializedInt", "SpecializedString"]
variance = "Contravariant"
