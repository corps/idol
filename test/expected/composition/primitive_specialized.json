{
  "SpecializedContravariantInt": {
    "enum": [],
    "fields": {},
    "is_a": [
      "any"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "SpecializedCovariantInt": {
    "enum": [],
    "fields": {},
    "is_a": [
      "primitive_specialized.SpecializedInt"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "SpecializedInt": {
    "enum": [],
    "fields": {},
    "is_a": [
      "int"
    ],
    "tags": [
      "thing"
    ],
    "trim": false,
    "variance": "Covariant"
  },
  "SpecializedString": {
    "enum": [],
    "fields": {},
    "is_a": [
      "string"
    ],
    "tags": [
      "thing"
    ],
    "trim": false,
    "variance": "Covariant"
  },
  "TwoSpecializedContravariant": {
    "enum": [],
    "fields": {},
    "is_a": [
      "any"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "UnspecializedInt": {
    "enum": [],
    "fields": {},
    "is_a": [
      "primitive_widens.IntAlias"
    ],
    "tags": [
      "description:hello"
    ],
    "trim": false,
    "variance": "Covariant"
  }
}
